pkg/kf/commands/install/gke.go

```diff
@@ -0,0 +1,908 @@
+		return nil, err
+	}
+
+	var lines []string
+	scanner := bufio.NewScanner(bytes.NewReader(output))
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+
+	return lines, scanner.Err()
+}
+
+func installServiceCatalog(ctx context.Context) error {
+	ctx = setLogPrefix(ctx, "Service Catalog")
+	Logf(ctx, "installing Service Catalog")
+	Logf(ctx, "downloading service catalog templates")
+	tempDir, err := ioutil.TempDir("", "kf-service-catalog")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		Logf(ctx, "cleaning up %s", tempDir)
+		os.RemoveAll(tempDir)
+	}()
+
+	tmpKfPath := path.Join(tempDir, "kf")
+
+	if _, err := git(
+		ctx,
+		"clone",
+		"https://github.com/google/kf",
+		tmpKfPath,
+	); err != nil {
+		return err
+	}
+
+	Logf(ctx, "applying templates")
+	if _, err := kubectl(
+		ctx,
+		"apply",
+		"-R",
+		"--filename", path.Join(tmpKfPath, "third_party/service-catalog/manifests/catalog/templates"),
+	); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func setupSpace(ctx context.Context, projID string) error {
+	ctx = setLogPrefix(ctx, "kf setup")
+	ok, err := selectYesNo(ctx, "Setup kf space?")
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return nil
+	}
+
+	Logf(ctx, "Setting up kf space")
+	spaceName, err := namePrompt(ctx, "Space Name: ", randName("space-"))
+	if err != nil {
+		return err
+	}
+	domain, err := hostnamePrompt(ctx, "Domain: ", "example.com")
+	if err != nil {
+		return err
+	}
+
+	if _, err := kf(
+		ctx,
+		"create-space", spaceName,
+		"--domain", domain,
+		"--container-registry", fmt.Sprintf("gcr.io/%s", projID),
+	); err != nil {
+		return err
+	}
+
+	if _, err := kf(ctx, "target", "-s", spaceName); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// gcloud will run the command and block until its done.
+func gcloud(ctx context.Context, args ...string) ([]string, error) {
+	return command(ctx, "gcloud", args...)
+}
+
+// kubectl will run the command and block until its done.
+func kubectl(ctx context.Context, args ...string) ([]string, error) {
+	return command(ctx, "kubectl", args...)
+}
+
+// kf will run the command and block until its done.
+func kf(ctx context.Context, args ...string) ([]string, error) {
+	return command(ctx, "kf", args...)
+}
+
+// git will run the command and block until its done.
+func git(ctx context.Context, args ...string) ([]string, error) {
+	return command(ctx, "git", args...)
+}
+
+type loggerType struct{}
+
+func setContextOutput(ctx context.Context, out io.Writer) context.Context {
+	return context.WithValue(ctx, loggerType{}, out)
+}
+
+type loggerPrefixType struct{}
+
+func setLogPrefix(ctx context.Context, prefix string) context.Context {
+	return context.WithValue(ctx, loggerPrefixType{}, prefix)
+}
+
+type verboseType struct{}
+
+func setVerbosity(ctx context.Context, verbose bool) context.Context {
+	return context.WithValue(ctx, verboseType{}, verbose)
+}
+
+func Logf(ctx context.Context, v string, args ...interface{}) {
```

Comment:
nit: this should be made private

Reply:
I'm actually going to move all these utility functions to a utils package under installer. This will be useful for other installers (e.g., minikube).